plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -3548,7 +3550,117 @@ public boolean destroyNetworkRulesForVM(final Connect conn, final String vmName)
         return true;
     }
 
-    public boolean defaultNetworkRules(final Connect conn, final String vmName, final NicTO nic, final Long vmId, final String secIpStr) {
+    /**
+     * Function to destroy the security group rules applied to the nic's
+     * @param conn
+     * @param vmName
+     * @param nic
+     * @return
+     *      true   : If success
+     *      false  : If failure
+     */
+    public boolean destroyNetworkRulesForNic(final Connect conn, final String vmName, final NicTO nic) {
+        if (!_canBridgeFirewall) {
+            return false;
+        }
+        final List<String> nicSecIps = nic.getNicSecIps();
+        String secIpsStr;
+        final StringBuilder sb = new StringBuilder();
+        if (nicSecIps != null) {
+            for (final String ip : nicSecIps) {
+                sb.append(ip).append(SecurityGroupRulesCmd.RULE_COMMAND_SEPARATOR);
+            }
+            secIpsStr = sb.toString();
+        } else {
+            secIpsStr = "0" + SecurityGroupRulesCmd.RULE_COMMAND_SEPARATOR;
+        }
+        final List<InterfaceDef> intfs = getInterfaces(conn, vmName);
+        if (intfs.size() == 0 || intfs.size() < nic.getDeviceId()) {
+            return false;
+        }
+
+        final InterfaceDef intf = intfs.get(nic.getDeviceId());
+        final String brname = intf.getBrName();
+        final String vif = intf.getDevName();
+
+        final Script cmd = new Script(_securityGroupPath, _timeout, s_logger);
+        cmd.add("destroy_network_rules_for_vm");
+        cmd.add("--vmname", vmName);
+        if (nic.getIp() != null) {
+            cmd.add("--vmip", nic.getIp());
+        }
+        cmd.add("--vmmac", nic.getMac());
+        cmd.add("--vif", vif);
+        cmd.add("--nicsecips", secIpsStr);
+
+        final String result = cmd.execute();
+        if (result != null) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Function to apply default network rules for a VM
+     * @param conn
+     * @param vm
+     * @param checkBeforeApply
+     * @return
```

Comment:
Thanks for adding documentation.

**Just a note:** I don't think that `@param` adds much info when it holds only the parameter name, normally I would vote to remove them, or add some description on each param.